CMS-Enterprise/mint-app
# Mint 3226/edit milestone links merge statement

# MINT-3226

## Description

Exploring splitting insert and merge statement
1. This results in shared reusable code for all milestone linking operations, instead of iterating through a list, or separate SQL calls.
2. This is now callable from the one `MTOMilestoneUpdateLinkedSolutionsWithTX` method.

Note we explored using a merge statement, which would be the ideal PGRES tool, however, the pq driver doesn't recognize the NOT MATCHED BY SOURCE syntax. As such, we went forward with the syntax @OddTomBrooks had already implemented in his pr, but broken out to it's own file.

## How to test this change

1.  Use the create common milestone to assert functionality remains unchanged
4. use the edit milestone mutation to assert functionality is as expected
5. use the link milestones mutation and assert it functions as expected.

## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
